src/ng/location.js

```diff
@@ -877,6 +879,8 @@ function $LocationProvider() {
         if (elm[0] === $rootElement[0] || !(elm = elm.parent())[0]) return;
       }
 
+      if (!html5Mode.rewriteLinks && isUndefined(elm.attr('ng-noserver'))) return;
```

Comment:
This will not recognize prefixed forms (such as `x-ng-...` or `data-ng-...`), but I am not sure this is an issue.